e2e/etcdctlv3_test.go

```diff
@@ -311,6 +398,119 @@ func ctlV3Watch(cx ctlCtx, key, value string) error {
 	return proc.Close()
 }
 
+type txnType int
+
+const (
+	versionCompReq txnType = iota
+	createCompReq
+	modCompReq
+	valueCompReq
+	getReq
+	putReq
+	deleteReq
+)
+
+type request struct {
+	reqType txnType
+	comp    string // comparison operation between key and value
+	kv      kv
+}
+
+type txnRequests struct {
+	compare  []request
+	ifSucess []request
+	ifFail   []request
+	results  []string
+}
+
+func (r request) String() string {
+	if r.reqType == versionCompReq ||
+		r.reqType == createCompReq ||
+		r.reqType == modCompReq ||
+		r.reqType == valueCompReq {
+		switch r.comp {
+		case "<":
+		case ">":
+		case ">=":
+		case "<=":
+		case "=":
+		default:
+			log.Fatalf("%q is unknown", r.comp)
+		}
+	}
+	switch r.reqType {
+	case versionCompReq:
+		return fmt.Sprintf("version(%q) %s %q", r.kv.key, r.comp, r.kv.val)
+	case createCompReq:
+		return fmt.Sprintf("create(%q) %s %q", r.kv.key, r.comp, r.kv.val)
+	case modCompReq:
+		return fmt.Sprintf("mod(%q) %s %q", r.kv.key, r.comp, r.kv.val)
+	case valueCompReq:
+		return fmt.Sprintf("value(%q) %s %q", r.kv.key, r.comp, r.kv.val)
+	case getReq:
+		return fmt.Sprintf("get %s", r.kv.key)
+	case putReq:
+		return fmt.Sprintf(`put %s %q`, r.kv.key, r.kv.val)
+	case deleteReq: // TODO: delete doesn't return the number of deleted keys yet
+		return fmt.Sprintf("delete %s", r.kv.key)
+	}
+	return ""
+}
+
+func ctlV3Txn(cx ctlCtx, rqs txnRequests) error {
+	// TODO: support non-interactive mode
+	cmdArgs := append(ctlV3PrefixArgs(cx.epc, cx.dialTimeout), "txn")
+	if cx.interactive {
+		cmdArgs = append(cmdArgs, "--interactive")
+	}
+	proc, err := spawnCmd(cmdArgs)
+	if err != nil {
+		return err
+	}
+	_, err = proc.Expect("compares:")
+	if err != nil {
+		return err
+	}
+	for _, req := range rqs.compare {
+		if err = proc.SendLine(req.String()); err != nil {
+			return err
+		}
+	}
+
+	_, err = proc.Expect("success requests (get, put, delete):")
```

Comment:
`SendLine` sounds broken since that means there's no way to test multiple ops per if/then/else block; the `\r\n` is probably being treated as two newlines instead of one. Maybe replace with `Send` and do newlines manually?